src/stream/stream/count.rs

```diff
@@ -0,0 +1,41 @@
+use std::pin::Pin;
+
+use crate::future::Future;
+use crate::stream::Stream;
+use crate::task::{Context, Poll};
+
+#[doc(hidden)]
+#[allow(missing_debug_implementations)]
+pub struct CountFuture<S> {
+    stream: S,
+    count: usize,
+}
+
+impl<S> CountFuture<S> {
+    pin_utils::unsafe_pinned!(stream: S);
+    pin_utils::unsafe_unpinned!(count: usize);
+
+    pub(crate) fn new(stream: S) -> Self {
+        CountFuture { stream, count: 0 }
+    }
+}
+
+impl<S> Future for CountFuture<S>
+where
+    S: Sized + Stream,
```

Comment:
I believe `S: Sized` is redundant here, i.e. `S` is assumed to be `Sized` by default unless we opt-out with `S: ?Sized`.

```suggestion
    S: Stream,
```

Reply:
Actually I don't remember why I used it in the first place. I think the compiler was complaining about it at some point. I actually don't see anymore why it should be `Sized` in the first place.

Will remove it.